backend/open_webui/models/functions.py

```
@@ -105,34 +131,137 @@ def insert_new_function(
+                for func in existing_functions:
+                    if func.id not in new_function_ids:
+                        db.delete(func)
+
+                db.commit()
+
+                return [
+                    FunctionModel.model_validate(func)
+                    for func in db.query(Function).all()
+                ]
+        except Exception as e:
+            log.exception(f"Error syncing functions for user {user_id}: {e}")
```

Ruff (0.15.4)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)